txrelayer/txrelayer.go

```diff
@@ -182,7 +182,9 @@ func (t *TxRelayerImpl) sendTransactionLocked(txn *ethgo.Transaction, key ethgo.
 	if txn.Gas == 0 {
 		gasLimit, err := t.client.Eth().EstimateGas(ConvertTxnToCallMsg(txn))
 		if err != nil {
-			return ethgo.ZeroHash, fmt.Errorf("failed to estimate gas: %w", err)
+			// TODO: Add Logger and write the err return ethgo.ZeroHash, fmt.Errorf("failed to estimate gas: %w", err)
+			// return ethgo.ZeroHash, fmt.Errorf("failed to estimate gas: %w", err)
+			gasLimit = DefaultGasLimit * 2
```

Comment:
I'm afraid that there would be the cases, where even twice the `DefaultGasLimit` would not help, because it is probably related to SC logic or other transaction parameters, so even if we set `gasLimit` to some (reasonable) high value, the transaction would still fail afterwards. So IMO I'd prefer to revert the old behavior.